packages/astro/src/cli/info/index.ts

```diff
@@ -12,31 +15,82 @@ interface InfoOptions {
 	flags: Flags;
 }
 
+export const VERSION_PREFIX_REGEX = /[\^~<>=]/g;
+
+type MinimalPackageJSON = {
+	version: string;
+	dependencies: Record<string, string>;
+}
+
+function getVersion(packageJsonPath: string, dependency?: string): string | undefined {
+	try {
+		const resolvedPath = createRequire(import.meta.url).resolve(packageJsonPath);
```

Comment:
This resolve won't work on strict package managers like pnpm, the `astro` package can only import its own dependencies, not dependencies of the root package.